src/emit/expression.rs

```diff
@@ -1904,106 +1904,182 @@ fn runtime_cast<'a>(
     val: BasicValueEnum<'a>,
     ns: &Namespace,
 ) -> BasicValueEnum<'a> {
-    if matches!(from, Type::Address(_) | Type::Contract(_))
-        && matches!(to, Type::Address(_) | Type::Contract(_))
-    {
-        // no conversion needed
-        val
-    } else if let Type::Address(_) = to {
-        let llvm_ty = bin.llvm_type(from, ns);
+    match (from, to) {
+        (Type::Address(_) | Type::Contract(_), Type::Address(_) | Type::Contract(_)) => {
+            // no conversion needed
+            val
+        }
```

Comment:
Why this match arm when the catch-all does the same?

Reply:
You're right, it's not necessary. Removed.